recipes/libgpiod/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conans.errors import ConanInvalidConfiguration
+import os
+import platform
+import re
+
+required_conan_version = ">=1.33.0"
+
+class LibgpiodConan(ConanFile):
+    name = "libgpiod"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/"
+    license = "LGPL-2.1-or-later"
+    description = "C library and tools for interacting with the linux GPIO character device"
+    topics = ("conan", "gpio", "libgpiod", "libgpiodcxx", "linux")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "enable_bindings_cxx": [True, False],
+               "enable_tools": [True, False]}
+    default_options = {"shared": False,
+                       "enable_bindings_cxx": False,
+                       "enable_tools": False}
+    
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("only Linux is supported")
+        linux_kernel_version = re.match("([0-9.]+)", platform.release()).group(1)
+        if tools.Version(linux_kernel_version) < "4.8":
+            raise ConanInvalidConfiguration("only Linux kernel versions >= 4.8 are supported")
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+        self.build_requires("pkgconf/1.7.4")
+        self.build_requires("autoconf-archive/2021.02.19")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = []
+            if self.options.shared:
+                args.extend(["--disable-static", "--enable-shared"])
+            else:
+                args.extend(["--disable-shared", "--enable-static"])
+            if self.options.enable_bindings_cxx:
+                args.extend(["--enable-bindings-cxx"])
+            if self.options.enable_tools:
+                args.extend(["--enable-tools"])
+            self._autotools = AutoToolsBuildEnvironment(self)
+            self._autotools.configure(args=args)
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), run_environment=True)
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            autotools = self._configure_autotools()
+            autotools.install()
+        tools.remove_files_by_mask(self.package_folder, "*.la")
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
Please add the names of the `pkg_config` generator in  `package_info`.
This allows conan to regenerate the `.pc` files when a dependency needs it.

You might need to use components.

Reply:
Will try to do this next

Reply:
@madebr the name is "libgpiod.pc" which matches the package name. Do we still need to add it explicitly?

Reply:
Yes.
The rationale is that, when it is added explicitly, somebody verified that the project creates a `.pc` file itself.
There is a possilbitity that the `pkg_config` generator will not generate `.pc` files for projects that does not set its name.

Reply:
Does it also create a `.pc` file for `gpiodcxx`? (`gpiodcxx.pc`)

Reply:
Also, when using componenent (as I do in my suggestion), you have to set the *names* attribute.

Reply:
Thanks @madebr . I'll look at this after this PR is unblocked.